wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,220 @@
+<?php
+/**
+ * WP_HTTPS_UI class.
+ *
+ * @package PWA
+ */
+
+/**
+ * WP_HTTPS_UI class.
+ */
+class WP_HTTPS_UI {
+
+	/**
+	 * The option group for the offline page option.
+	 *
+	 * @var string
+	 */
+	const OPTION_GROUP = 'reading';
+
+	/**
+	 * The option name to upgrade to https.
+	 *
+	 * @var string
+	 */
+	const UPGRADE_HTTPS_OPTION = 'should_upgrade_https';
+
+	/**
+	 * The option name to ugprade insecure content.
+	 *
+	 * @var string
+	 */
+	const UPGRADE_INSECURE_CONTENT_OPTION = 'should_upgrade_insecure_content';
+
+	/**
+	 * The expected value if the option is enabled.
+	 *
+	 * @var string
+	 */
+	const OPTION_SELECTED_VALUE = '1';
+
+	/**
+	 * The ID of the settings section.
+	 *
+	 * @var string
+	 */
+	const SETTING_ID = 'wp_upgrade_https';
+
+	/**
+	 * The HTTPS protocol
+	 *
+	 * @var string
+	 */
+	const HTTPS_PROTOCOL = 'https://';
+
+	/**
+	 * Inits the class.
+	 */
+	public function init() {
+		add_action( 'admin_init', array( $this, 'init_admin' ) );
+		add_action( 'admin_init', array( $this, 'filter_site_url_and_home' ) );
+	}
+
+	/**
+	 * Initializes the admin tasks.
+	 */
+	public function init_admin() {
+		$this->register_settings();
+		$this->add_settings_field();
+	}
+
+	/**
+	 * Registers the HTTPS settings.
+	 */
+	public function register_settings() {
+		$args = array(
+			'type'              => 'string',
+			'sanitize_callback' => array( $this, 'sanitize_callback' ),
+		);
+
+		register_setting(
+			self::OPTION_GROUP,
+			self::UPGRADE_HTTPS_OPTION,
+			$args
+		);
+
+		register_setting(
+			self::OPTION_GROUP,
+			self::UPGRADE_INSECURE_CONTENT_OPTION,
+			$args
+		);
+	}
+
+	/**
+	 * Sanitizes the raw setting value.
+	 *
+	 * @param string $raw_setting The setting before sanitizing it.
+	 * @return string|null The setting, or null if it's not valid.
+	 */
+	public function sanitize_callback( $raw_setting ) {
+		if ( self::OPTION_SELECTED_VALUE === $raw_setting || '0' === $raw_setting ) {
+			return $raw_setting;
+		}
```

Comment:
I think this should return `get_option( self::UPGRADE_HTTPS_OPTION )` if the sanitization fails. But then again, why not just use `wp_validate_boolean()` as the `sanitize_callback`?

Reply:
Sure, this could use `wp_validate_boolean` as the `sanitize_callback`. It's probably a good solution, though it will return `true` for most strings, not just `'1'`.

The expected values here are `'1'` [if Yes is selected](https://github.com/xwp/pwa-wp/pull/36/files#diff-cd06676b91b7dea56c702cb7bc9e7ec9R166), and `'0'` [if No is selected](https://github.com/xwp/pwa-wp/pull/36/files#diff-cd06676b91b7dea56c702cb7bc9e7ec9R167). 

That validation callback would return `true` for almost any string other than `'0'` or `'false'`, and `false` for the string `'0'`.

Reply:
Yeah, that should be fine.

Reply:
Thanks, [https://github.com/xwp/pwa-wp/pull/36/commits/c2cbdeea2902ffcbaa94bdc21a857dacb16f43bb](https://github.com/xwp/pwa-wp/pull/36/commits/c2cbdeea2902ffcbaa94bdc21a857dacb16f43bb) passes `wp_validate_boolean` as the `sanitize_callback`.